pkg/adapter/adapter.go

```diff
@@ -166,12 +167,38 @@ func (s *sinker) getCertFromClusterInterceptor(certPool *x509.CertPool) error {
 				}
 			}
 		}
-		if httpsCILen != 0 && httpsCILen == count {
-			return true, nil
+
+		if httpsCILen == 0 || httpsCILen != count {
+			return false, fmt.Errorf("empty caBundle in clusterInterceptor spec")
 		}
-		return false, fmt.Errorf("empty caBundle in clusterInterceptor spec")
+
+		httpsCILen = 0
+		count = 0
+
+		interceptorList, err := interceptorsinformer.Get(s.injCtx).Lister().Interceptors(s.Namespace).List(labels.NewSelector())
```

Comment:
@dibyom  Error is coming from this line. It happens during startup.

```
{"severity":"fatal","timestamp":"2022-10-19T18:18:41.222Z","logger":"eventlistener","caller":"v2/main.go:205","message":"Start returned an error","commit":"142a5be","error":"Timed out waiting on CaBundle to available for clusterInterceptor: Timed out waiting on CaBundle to available for Interceptor: resource triggers.tekton.dev/v1alpha1, Resource=interceptors not supported"}
```

Reply:
This error happens if caBundle absent in the interceptor 

But how come your changes effecting this :thinking:

Reply:
It seems like somewhere the interceptor resource is not registered - I'm guessing something related to the informer being set up